lib/public/Federation/ICloudFederationProviderManager.php

```
@@ -80,9 +83,18 @@ public function getCloudFederationProvider($resourceType);
 	 * @return mixed
 	 *
 	 * @since 14.0.0
+	 * @deprecated 29.0.0 - Use {@see sendCloudShare()} instead and handle errors manually
 	 */
 	public function sendShare(ICloudFederationShare $share);
 
+	/**
+	 * @param ICloudFederationShare $share
+	 * @return IResponse
+	 * @throws OCMProviderException
+	 * @since 29.0.0
+	 */
+	public function sendCloudShare(ICloudFederationShare $share): IResponse;
```

why add two new methods when you could modify the exisitng one?

---

because it would now throw instead of returning false, which would break the current consumers?